fixed #155: back to top button

mansoorbarri opened the pull request:
Fixes #155

Screenshot: 
![image](https://github.com/526avijitgupta/gokarna/assets/90044118/360df791-b165-4e43-9dce-b11bfbe0f7ac)

yashmehrotra commented:
@mansoorbarri Thanks for the PR

Can you also:
1. Feature gate it with the `ShowBackToTopButton` param
2. Add the feature in theme-basic-documentation

mansoorbarri commented:
fixed! in 

> @mansoorbarri Thanks for the PR
> 
> Can you also:
> 
> 1. Feature gate it with the `ShowBackToTopButton` param
> 2. Add the feature in theme-basic-documentation

fixed

526avijitgupta commented on assets/css/main.css:
```diff
@@ -624,3 +624,22 @@ table td {
+
+/* back to top button  */
+.arrow-logo {
+  display: none; /* Hidden by default */
+  position: fixed; /* Fixed/sticky position */
+  bottom: 20px; /* Place the icon at the bottom of the page */
+  right: 30px; /* Place the icon 30px from the right */
+  z-index: 99; /* Make sure it does not overlap */
+  background-color: transparent; /* Remove the background color */
+  color: var(--accent-color); /* Icon color */
+  cursor: pointer; /* Add a mouse pointer on hover */
+  padding: 10px; /* Adjust the padding as needed */
+  border-radius: 50%; /* Make it a circle */
+  font-size: 24px; /* Increase font size */
```
1. please remove comments
2. use z-index `5` (to match the `z-index` value of header element)
3. order alphabetically

```css
.arrow-logo {
  background-color: transparent;
  border-radius: 50%;
  bottom: 20px;
  color: var(--accent-color);
  cursor: pointer;
  display: none;
  font-size: 24px;
  padding: 10px;
  position: fixed;
  right: 30px;
  z-index: 5;
}
```

526avijitgupta reviewed (commented):
Hi @mansoorbarri thanks for submitting the PR. This feature would be quite useful :)

While the feature currently works, it abruptly takes the user to the top of the page. I would prefer if there is a smooth srolling to the top of the page. What I mean is it should work similar to how clicking on an item in the Table of Contents (on the right side of the page) works. @yashmehrotra wdyt?

Here's a code reference of how to smooth scrolling is implemented currently: https://github.com/526avijitgupta/gokarna/blob/be5611d4f02d679b43e869c1a542ecbf84f0febc/assets/js/main.js#L68C1-L78C2

mansoorbarri commented:
appreciate the feedback. Please have a look now

yashmehrotra reviewed (approved):
LGTM 👍🏽 

@526avijitgupta Can you have a final look and merge this ?

526avijitgupta commented:
@mansoorbarri This seems to break on mobile. . Could you please double check on your side and fix it before we merge?

Otherwise if someone builds Gokarna on the latest version and tries to use it on mobile screens it wouldn't be a good user experience for them

Also, let's use an icon which doesn't have a vertical line at at all. Basically, similar to an inverted "V"

yashmehrotra commented:
Yeah, we should either pad it in way it does not come over text or disable it for mobile

![image](https://github.com/526avijitgupta/gokarna/assets/5863972/6653df03-eb9b-4c3e-a0b1-5f1c5c640e1a)

mansoorbarri commented:
I find this a bit alright but removing the button on mobile is probably the best option in my opinion. Do let me know what you would like.

preview: 
| Dark         | Light          |
| ------------------ | ------------------ |
| ![image](https://github.com/526avijitgupta/gokarna/assets/90044118/7cd195e1-c0e7-4212-a32e-72e72f3b1926) | ![image](https://github.com/526avijitgupta/gokarna/assets/90044118/1d9e8bf5-48f5-4453-bd1f-53c0c46a69eb)|

526avijitgupta commented:
@mansoorbarri Thanks for making the changes.
In the current state the button looks okay, but we can certainly make it more visually appealing by making it match the site theme (light/dark).

I was browsing Hugo themes and came across this as an example: https://hugoloveit.com/emoji-support/#

Notice how the button's color and background color matches the current theme (light/dark).

![Screenshot_20231026-233909.png](https://github.com/526avijitgupta/gokarna/assets/7583815/93868236-c738-4d8e-8f1d-ac1bbef1b0bb)

![Screenshot_20231026-233934.png](https://github.com/526avijitgupta/gokarna/assets/7583815/b0417ae6-2eac-4601-b7c5-8e3c386f7218)

526avijitgupta commented:
Basically, what I mean is the blue background color on the back to top button looks a little out of place IMO. Simply changing that to match the light/dark theme should also be enough :)

mansoorbarri commented:
definitely, my first idea was to make it the same colour as code blocks on the website but I wasn't sure about that. 

preview: https://gokarna.mansoorbarri.com

they look something like this: 
| Dark         | Light      |
| ------------------ | ------------------ |
| ![image](https://github.com/526avijitgupta/gokarna/assets/90044118/cffccbb6-2f9e-4045-943c-0046c093f2c7) | ![image](https://github.com/526avijitgupta/gokarna/assets/90044118/dc0a8c06-8369-4e6f-a192-f095ca75b070) |

mansoorbarri commented:
> definitely, my first idea was to make it the same colour as code blocks on the website but I wasn't sure about that.
> 
> preview: https://gokarna.mansoorbarri.com
> 
> they look something like this:
> 
> Dark	Light
> ![image](https://user-images.githubusercontent.com/90044118/278484666-cffccbb6-2f9e-4045-943c-0046c093f2c7.png)	![image](https://user-images.githubusercontent.com/90044118/278484708-dc0a8c06-8369-4e6f-a192-f095ca75b070.png)

we can also make it a bit opaque but if you have any other ideas, do let me know.

yashmehrotra commented:
@mansoorbarri @526avijitgupta 

I think we should disable this on mobile and create a different issue to track that till we come up with a better design.
Lets try to get this PR merged with mobile disabled

What do you all think ?

mansoorbarri commented:
I agree

mansoorbarri commented:
the most we can do is move the button a bit towards right but it still comes over some of the page content: 
![image](https://github.com/526avijitgupta/gokarna/assets/90044118/8e954eaf-42da-42e5-aef6-999465e938fe)

yashmehrotra commented:
> the most we can do is move the button a bit towards right but it still comes over some of the page content:

@mansoorbarri Yeah, that is why I don't want to add this in mobile till we come up with a better solution

But I like adding a background color with the arrow

Lets:
1. Disable it for mobile
2. Add background color with arrow with just the inverted v

526avijitgupta commented:
> @mansoorbarri @526avijitgupta
> 
> I think we should disable this on mobile and create a different issue to track that till we come up with a better design.

@yashmehrotra Agreed!

mansoorbarri commented:
> > the most we can do is move the button a bit towards right but it still comes over some of the page content:
> 
> @mansoorbarri Yeah, that is why I don't want to add this in mobile till we come up with a better solution
> 
> But I like adding a background color with the arrow
> 
> Lets:
> 
> 1. Disable it for mobile
> 2. Add background color with arrow with just the inverted v

added

yashmehrotra commented:
![image](https://github.com/526avijitgupta/gokarna/assets/5863972/184f3ed8-6805-4b8d-b284-db3cfdf9009d)

Can we make the padding in such a way that the V is in the middle, right now its left-centered

mansoorbarri commented:
I tried multiple changes to the CSS code however it still stays left aligned, the `svg` doesn't have a seperate div so I don't know why it shows up like that

yashmehrotra commented on assets/css/main.css:
```diff
@@ -624,3 +624,33 @@ table td {
+  right: 20px;
+  z-index: 5;
+  background-color: var(--light-tertiary-color);
+  border-radius: 10px;
```
Border radius is twice, lets make this a complete circle

yashmehrotra commented:
@mansoorbarri Doing `padding-right: 8px` makes it look like its at the centre
Also, lets make this a complete circle

mansoorbarri commented:
@yashmehrotra all changes are made, preview: https://gokarna.mansoorbarri.com/

yashmehrotra commented:
@mansoorbarri Looks good

@526avijitgupta Should we merge this PR now ?
